pkg/cmd/join/exec.go

```diff
@@ -92,6 +97,16 @@ func (o *Options) validate() error {
 		return fmt.Errorf("registry should not be empty")
 	}
 
+	if o.mode != "default" && o.mode != "hosted" {
```

Comment:
should set a default value, we should not let user to always set mode

Reply:
yes, the default value is `default` https://github.com/open-cluster-management-io/clusteradm/pull/289/files#diff-af2f69ba48e16b856a2cce7544ec645881c15d4840f8922fcfb0e49505fb1e39R58